### PR TITLE
Authorization refactor

### PR DESCRIPTION
**Pull request by sbathgate**
This PR refactors how the registry conducts its Authorization processes. The previous system relied exclusively upon the signed in users KeyCloak id, this refactor enables a users email to be a means of authorization also.

### CODE REVIEW

File: api/src/controllers/profile.ts [hunk 1]
```diff
@@ -273,21 +272,19 @@ export const addContactToProfile = async (
...
-  const { ContactModel, ProfileModel } = dm;
+  const { ContactModel } = dm;
   const { profileId } = params;
 
   try {
-    const record = await ProfileModel.findById(Number(profileId));
+    const projectContacts = await ContactModel.findForProject(Number(profileId));
 
-    const notAuthorized = isNotAuthorized(record, user);
+    const isAuthorized = getAuthorization(profileId, user, undefined, projectContacts);
```
Comment (yhfreeman) [inline api/src/controllers/profile.ts:281]: Just nitpick on readability. Currently getAuthorization either throw an Error or return true. It's just a bit odd to "throw isAuthorized" when "not authorized"?

File: api/src/controllers/profile.ts [hunk 2]
```diff
@@ -83,15 +84,15 @@ export const fetchProjectProfile = async (
...
+    const projectDetails = await ProfileModel.findById(Number(profileId));
 
-    const notAuthorized = isNotAuthorized(record, user);
+    const isAuthorized = getAuthorization(profileId, user, projectDetails, undefined);
 
-    if (notAuthorized) {
-      throw notAuthorized;
+    if (!(isAuthorized)) {
+      throw isAuthorized;
     }
```
Comment (yhfreeman) [inline api/src/controllers/profile.ts:93]: Just nitpick on readability. Currently getAuthorization either throw an Error or return true. It's just a bit odd to "throw isAuthorized" when "not authorized"?

**Review by yhfreeman — APPROVED** [review]
The feature is functioning as expected - thanks for your work. Perhaps libs/authorization could somehow be be combined with libs/authware. Also, would it be worth cleaning up some linter issues according to sonarcloud?

File: api/src/controllers/profile.ts (hunk 2 above)
Comment (sbathgate) [reply api/src/controllers/profile.ts:93]: Absolutely agree, I have adjusted to isNotAuthorized to reflect the logic more appropriately.

**Pull request closed without merging**